### DISCUSSION TITLE
ImbalancedSampler Error

### DISCUSSION THREAD

**Discussion by 1AngelUS** [pyg-team/pytorch_geometric#9383] [General]
Hi,

I am using ImbalancedSampler for hetero data:
sampler = ImbalancedSampler(data['p'].y.long())
train_loader = NeighborLoader(data, batch_size = 500, num_neighbors=[20,10]*2,input_nodes=train_input_nodes,
                              sampler=sampler)

This is throwing the following error:
RuntimeError: number of categories cannot exceed 2^24

**Comment by rusty1s** [comment]
How big is your `y` tensor? PyTorch currently has a limitation of max 2^24 samples :(
